### PR TITLE
Topology update for every Nth id volume

### PR DESCRIPTION
**Pull request by kmlebedev**
https://github.com/chrislusf/seaweedfs/issues/1846

### CODE REVIEW

File: weed/shell/command_volume_fix_replication.go [hunk 1]
```diff
@@ -56,6 +56,8 @@ func (c *commandVolumeFixReplication) Do(args []string, commandEnv *CommandEnv,
 	c.collectionPattern = volFixReplicationCommand.String("collectionPattern", "", "match with wildcard characters '*' and '?'")
 	skipChange := volFixReplicationCommand.Bool("n", false, "skip the changes")
 	retryCount := volFixReplicationCommand.Int("retry", 0, "how many times to retry")
+	volumespPerStep := volFixReplicationCommand.Int("volumes_per_step", 0, "how many volumes to fix in one cycle")
```
Comment (chrislusf) [inline weed/shell/command_volume_fix_replication.go]: The variable name:
volumespPerStep => volumesPerStep

To be consistent with other parameters:
volumes_per_step => volumesPerStep

**Comment by chrislusf** [comment]
Adding a loop may not be enough. The topology information was collected for each loop. But the recent topology changes may not have been updated back to the master yet. So if loop too quickly, there would be stale topology information, causing possible wrong fix operations.

For example, if one replication fix was applied but not reported to the master yet, the fix would be applied twice.

**Comment by kmlebedev** [comment]
> For example, if one replication fix was applied but not reported to the master yet, the fix would be applied twice.

On the fly, there is an idea at the end of the cycle to do a LookupVolume and make sure the information has been updated

**Comment by kmlebedev** [comment]
> For example, if one replication fix was applied but not reported to the master yet, the fix would be applied twice.

It seems unlikely that I checked it on a volume of 10 mb

```
> volume.fix.replication -n
replicating volume 3 200 from volume3:8083 to dataNode volume2:8082 ...
replicating volume 4 200 from volume3:8083 to dataNode volume1:8080 ...
replicating volume 6 200 from volume3:8083 to dataNode volume2:8082 ...
replicating volume 7 200 from volume1:8080 to dataNode volume3:8083 ...
replicating volume 5 200 from volume1:8080 to dataNode volume3:8083 ...
replicating volume 8 200 from volume3:8083 to dataNode volume2:8082 ...
replicating volume 2 200 from volume3:8083 to dataNode volume2:8082 ...
replicating volume 9 200 from volume3:8083 to dataNode volume1:8080 ...
> volume.fix.replication -volumes_per_step 1
replicating volume 5 200 from volume1:8080 to dataNode volume3:8083 ...
replicating volume 3 200 from volume1:8080 to dataNode volume2:8082 ...
replicating volume 8 200 from volume3:8083 to dataNode volume2:8082 ...
replicating volume 9 200 from volume3:8083 to dataNode volume1:8080 ...
replicating volume 2 200 from volume1:8080 to dataNode volume2:8082 ...
replicating volume 6 200 from volume1:8080 to dataNode volume2:8082 ...
replicating volume 7 200 from volume2:8082 to dataNode volume3:8083 ...
replicating volume 4 200 from volume2:8082 to dataNode volume1:8080 ...
```

**Comment by chrislusf** [comment]
> > For example, if one replication fix was applied but not reported to the master yet, the fix would be applied twice.
> 
> It seems unlikely that I checked it on a volume of 10 mb
> 
> ```
> > volume.fix.replication -n
> replicating volume 3 200 from volume3:8083 to dataNode volume2:8082 ...
> replicating volume 4 200 from volume3:8083 to dataNode volume1:8080 ...
> replicating volume 6 200 from volume3:8083 to dataNode volume2:8082 ...
> replicating volume 7 200 from volume1:8080 to dataNode volume3:8083 ...
> replicating volume 5 200 from volume1:8080 to dataNode volume3:8083 ...
> replicating volume 8 200 from volume3:8083 to dataNode volume2:8082 ...
> replicating volume 2 200 from volume3:8083 to dataNode volume2:8082 ...
> replicating volume 9 200 from volume3:8083 to dataNode volume1:8080 ...
> > volume.fix.replication -volumes_per_step 1
> replicating volume 5 200 from volume1:8080 to dataNode volume3:8083 ...
> replicating volume 3 200 from volume1:8080 to dataNode volume2:8082 ...
> replicating volume 8 200 from volume3:8083 to dataNode volume2:8082 ...
> replicating volume 9 200 from volume3:8083 to dataNode volume1:8080 ...
> replicating volume 2 200 from volume1:8080 to dataNode volume2:8082 ...
> replicating volume 6 200 from volume1:8080 to dataNode volume2:8082 ...
> replicating volume 7 200 from volume2:8082 to dataNode volume3:8083 ...
> replicating volume 4 200 from volume2:8082 to dataNode volume1:8080 ...
> ```

This is a data racing issue and may be possibly reproducible on larger clusters.

Verifying via a lookup seems a good idea.

**Comment by kmlebedev** [comment]
> Verifying via a lookup seems a good idea.

added

**Pull request merged**